File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        res = 0
        l = 0
        r = len(heights) - 1

        while r > l:
            temp = (r - l) * min(heights[l], heights[r])
            if temp > res:
                res = temp
            if heights[l] > heights[r]:
                r-=1
            else:
                l+=1


        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());

        int res = 0;

        for (int num : s){
            if (s.find(num-1) == s.end()){

                int current = num;
                int temp = 0;
                
                while (s.find(current) != s.end()){
                    current++;
                    temp++;
                    res = max(res, temp);
                }

            }

        }
        return res;
    }
};
